src/test/java/com/google/finapp/FinAppIT.java

```diff
@@ -19,67 +19,120 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 import com.google.cloud.ByteArray;
+import com.google.cloud.spanner.Database;
 import com.google.cloud.spanner.DatabaseClient;
-import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.IntegrationTest;
+import com.google.cloud.spanner.IntegrationTestEnv;
 import com.google.cloud.spanner.Key;
 import com.google.cloud.spanner.KeySet;
 import com.google.cloud.spanner.ResultSet;
-import com.google.cloud.spanner.Spanner;
-import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.testing.RemoteSpannerHelper;
 import java.math.BigDecimal;
 import java.util.Arrays;
+import java.util.List;
 import java.util.UUID;
-import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.ValueSource;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
 
+@Category(IntegrationTest.class)
 public class FinAppIT {
 
   private static SpannerDaoInterface JDBCDao;
   private static SpannerDaoInterface JavaDao;
   private static DatabaseClient databaseClient;
+  private static Database db;
 
-  static {
-    // set all IDs according to emulator setup
-    String spannerProjectId = "test-project";
-    String spannerInstanceId = "test-instance";
-    String spannerDatabaseId = "test-database";
-    JDBCDao = new SpannerDaoJDBCImpl(spannerProjectId, spannerInstanceId, spannerDatabaseId);
-    SpannerOptions spannerOptions = SpannerOptions.getDefaultInstance();
-    Spanner spanner = spannerOptions.toBuilder().build().getService();
-    databaseClient = spanner
-        .getDatabaseClient(DatabaseId.of(spannerProjectId, spannerInstanceId, spannerDatabaseId));
+  @ClassRule
+  public static IntegrationTestEnv env = new IntegrationTestEnv();
+
+  @BeforeClass
+  public static void setup() {
+    final RemoteSpannerHelper testHelper = env.getTestHelper();
+    db = testHelper.createTestDatabase(
```

Comment:
It is not ideal to directly copy from the schema file. However, I had trouble finding either Cloud Spanner Java API or `IntegrationTestEnv`-related methods for using an `sdl` file to set up our databases programmatically. Most examples I saw used Cloud Spanner CLI (i.e. `gcloud spanner databases create test-database --ddl-file src/main/java/com/google/finapp/schema.sdl`).

Please let me know if you have any ideas for accomplishing this setup in Java or if this looks okay.

Reply:
How about you load the schema from the file?

For example on how to do this, see here:
https://github.com/googleapis/java-spanner-jdbc/blob/6d545b617cd2ab21de88a77ce8e9ebe31cf088bf/src/main/java/com/google/cloud/spanner/jdbc/JdbcDatabaseMetaData.java#L51

btw, we should probably move the file in a resources folder, e.g: check out the files that are being loaded from the above:
https://github.com/googleapis/java-spanner-jdbc/tree/6d545b617cd2ab21de88a77ce8e9ebe31cf088bf/src/main/resources/com/google/cloud/spanner/jdbc

Reply:
+1 on putting all of this in the resources folder in its own file just for the purposes of testing.